src/Nethermind/Nethermind.Consensus/Processing/OverridableTxProcessingScope.cs

```diff
@@ -8,22 +8,34 @@
 
 namespace Nethermind.Consensus.Processing;
 
-public class OverridableTxProcessingScope(
-    IOverridableCodeInfoRepository codeInfoRepository,
-    ITransactionProcessor transactionProcessor,
-    IOverridableWorldState worldState,
-    Hash256 originalStateRoot
-) : IOverridableTxProcessingScope
+public class OverridableTxProcessingScope : IOverridableTxProcessingScope
 {
-    public IOverridableCodeInfoRepository CodeInfoRepository => codeInfoRepository;
-    public ITransactionProcessor TransactionProcessor => transactionProcessor;
-    public IWorldState WorldState => worldState;
+    private readonly IOverridableCodeInfoRepository _codeInfoRepository;
+    private readonly ITransactionProcessor _transactionProcessor;
+    private readonly IOverridableWorldState _worldState;
 
-    public void Dispose()
+    public OverridableTxProcessingScope(IOverridableCodeInfoRepository codeInfoRepository,
+        ITransactionProcessor transactionProcessor,
+        IOverridableWorldState worldState,
+        Hash256 stateRoot)
     {
-        worldState.StateRoot = originalStateRoot;
-        worldState.Reset();
-        worldState.ResetOverrides();
-        codeInfoRepository.ResetOverrides();
+        _codeInfoRepository = codeInfoRepository;
+        _transactionProcessor = transactionProcessor;
+        _worldState = worldState;
+        Reset();
```

Comment:
So we are resetting state before each RPC call now (instead of doing it after override-only calls)?
Any estimations on performance impact?

Reply:
Probably minimal, just some `Clears` on collections, I would prefer to reset it on Dispose but we probably have some use cases where we leak it, so doing it like that for safety.